### ISSUE TITLE
Boundary and Initial conditions directory structure of Aspect

### ISSUE THREAD

**Issue by gassmoeller** [geodynamics/aspect#179]
Considering that we already have a number of significant changes for the next release version of Aspect, I would like to propose one more rather large change. I find the current naming structure of source and include has developed in a suboptimal way for the boundary and initial condition plugins. We have three different naming conventions (1:boundary_composition/boundary_temperature, 2:initial_conditions, 3:compositional_initial_conditions/velocity_boundary_conditions) and especially when explaining the code to new users and introducing them into where to make changes this is kind of confusing.
I would personally consider naming convention 1 the best, and if the folders would be renamed that way, we would have the following folders:
boundary_composition, boundary_temperature, boundary_velocity, initial_temperature, initial_composition
I am aware that this is a change that introduces a lot of incompatibility (for user-written plugins that are not in master) and work (update many include commands - also in shared libraries for benchmarks and testcases) and should be considered carefully, before implementing it, but I also think that the improvement in structure and user-friendliness would be worth the effort. Especially considering that this change would not longer be possible, when we are in a later, more stable stage of the development. 
Please let me know if you find this idea unfeasible or unnecessary, I appreciate any comment.

**Comment by tjhei** [comment]
I vote for also redoing the parameter sections while we are at it and make it more flexible. I was thinking something like this:
```
subsection boundary conditions
  subsection Velocity
    set conditions = left: tangential, right: zero, top: function, bottom: SomePluginName
    subsection function
      set function = bla
    end
  end
  subsection Temperature
    set conditions = left: periodic, right: periodic, top: function, bottom: initial
end

subsection Initial conditions
  like above
end
```
The idea is that there is one list for the indicators and you can pick between all kind of plugins and different types of boundaries.

This has many advantages:
- it is more flexible (right now you can only pick a single model for temperature)
- easier to validate that each boundary has exactly one condition (in the code and for the user)
- it is the same for all types of boundaries

**Comment by gassmoeller** [comment]
Closed by #1449 and #1451.
